File: Week-1/Quick-Find/QuickFindUF.swift
//Class QuickFindUF in Swift
class QuickFindUF {
    var id: [Int] //declare a property call id
    
    init(N: Int)
    {
        id = [Int]() //create an empty array of id
        
        for i in 0...N {
            id[i] = i; //set id of each object to itself
        }
    }
    
    
    //Check to see if p and q have the same id
    func connected(p: Int, q: Int) -> Bool {
        return id[q] == id[q]
    }
    
    //Merge q and q
    func union(p: Int, q: Int) {
        let qid = id[q]
        let pid = id[p]
        
        for i in id {
            if id[i] == pid {
                id[i] = qid
            }
        }
    }
}

File: Week-1/Quick-Find/QuickFindUF.java
public class QuickFindUF
{
  private int[] id;
  
  //Default constructor
  public QuickFindUF(int N)
  {
    id = new int[N]; //init the array of N size
    for (int i = 0; i < N; i++) {
      id[i] = i; //set id of each object to itself
    }
  }
  
  public boolean connected(int p, int q) 
  {
    return id[p] == id[q];
  }
  
  public void union(int p, int q)
  {
    int pid = id[p];
    int qid = id[q];
    
    for (int i = 0; i < id.length(); i++)
    {
      if (id[i] == pid) id[i] = qid;
    }
  }
}
